filemanager/gcsmanager.go

```diff
@@ -28,18 +30,32 @@ type GCSConfig struct {
 	ForcePathStyle *bool
 	DisableSSL     *bool
 	JSONReads      bool
+
+	doNotOverWrite bool
+}
+
+type Opt func(*GCSConfig)
```

Comment:
We can make it specific to GCS options as these are specific to GCS configs.
```suggestion
type GCSOpt func(*GCSConfig)
```